commons/src/test/java/io/aiven/kafka/tiered/storage/commons/storage/filesystem/FileSystemStorageFactoryTest.java

```diff
@@ -0,0 +1,103 @@
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.aiven.kafka.tiered.storage.commons.storage.filesystem;
+
+import java.io.ByteArrayInputStream;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Path;
+import java.util.HashMap;
+import java.util.Map;
+
+import io.aiven.kafka.tiered.storage.commons.storage.ObjectStorageFactory;
+
+import org.apache.commons.io.FileUtils;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.NullSource;
+import org.junit.jupiter.params.provider.ValueSource;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatNoException;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+class FileSystemStorageFactoryTest {
+    @ParameterizedTest
+    @NullSource
+    @ValueSource(booleans = { false })
+    void testNotAllowOverwriting(final Boolean overwriteEnabled,
+                                 @TempDir final Path tempDir) throws IOException {
+        final File fsRoot = tempDir.toFile();
+        final ObjectStorageFactory osFactory = new FileSystemStorageFactory();
+        final HashMap<String, String> config = new HashMap<>();
+        config.put("root", fsRoot.toString());
+        if (overwriteEnabled != null) {
+            config.put("overwrite.enabled", Boolean.toString(overwriteEnabled));
+        }
+        osFactory.configure(config);
+
+        final InputStream logFile = new ByteArrayInputStream("log file".getBytes());
+        osFactory.fileUploader().upload(logFile, "aaa/0.log.txt");
+
+        assertThatThrownBy(() -> osFactory.fileUploader().upload(logFile, "aaa/0.log.txt"))
+            .isInstanceOf(IOException.class)
+            .hasMessage("File %s already exists", fsRoot + "/aaa/0.log.txt");
+    }
+
+    @Test
+    void testAllowOverwriting(@TempDir final Path tempDir) throws IOException {
+        final File fsRoot = tempDir.toFile();
+        final ObjectStorageFactory osFactory = new FileSystemStorageFactory();
+        osFactory.configure(Map.of(
+            "root", fsRoot.toString(),
+            "overwrite.enabled", "true"
+        ));
+
+        final InputStream logFile = new ByteArrayInputStream("log file".getBytes());
+        osFactory.fileUploader().upload(logFile, "aaa/0.log.txt");
+        assertThatNoException().isThrownBy(() -> osFactory.fileUploader().upload(logFile, "aaa/0.log.txt"));
```

Comment:
What do we actually test here? The assertion is not really necessary I guess since the exception will fail the test anyways and we do not verify that the file was actually overwritten.

Reply:
I think the whole purpose of `assertThatNoException` is informational, kind of instead of `// no exception should be thrown here`.

Reply:
Discussed offline, fixed